packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -121,17 +121,39 @@ abstract class RenderSliverBoxChildManager {
   /// true without making any assertions.
   bool debugAssertChildListLocked() => true;
 }
-
-/// Parent data structure used by [RenderSliverMultiBoxAdaptor].
-class SliverMultiBoxAdaptorParentData extends SliverLogicalParentData with ContainerParentDataMixin<RenderBox> {
-  /// The index of this child according to the [RenderSliverBoxChildManager].
-  int index;
+/// Parent data structure used by [RenderSliverWithKeepAliveMixin].
+abstract class KeepAliveParentDataMixin implements ParentData {
```

Comment:
Should this also just be `mixin KeepAliveParentDataMixin`?  Then we could drop the private `factory` constuctor here.